galsim/utilities.py

```diff
@@ -418,6 +418,103 @@ def _gammafn(x):
+    a lower sampling scale (or higher sampling rate) using the `pixel_scale' argument in drawImage()
+    routine and setting the `method' parameter to `no_pixel'.
+
+    Here is an example script using this routine:
+
+    Interleaving four Gaussian images
+    ---------------------------------
+
+        >>> n = 2
+        >>> gal = galsim.Gaussian(sigma=2.8)
+        >>> gal_pix = galsim.Convolve([gal,galsim.Pixel(scale=1.0)])
+        >>> img = gal_pix.drawImage(gal_pix,scale=1.0*n,method='no_pixel')
+        >>> im_list, offsets = galsim.utilities.deInterleaveImage(gal_pix,N=n)
+        >>> for im in im_list:
+        >>>     im.applyNonlinearity(lambda x: x-0.01*x**2) #detector effects
+        >>> img_new = galsim.utilities.interleaveImages(im_list,N=n,offsets)
+
+    @param image             Input image from which lower resolution images are extracted.
+    @param N                 Number of images extracted in either directions. It can be of type
+                             'int' if equal number of images are extracted in both directions or a
+                             list or tuple of two integers, containing the number of images in x
+                             and y directions respectively.
+    @param conserve_flux     Should the routine output images that have, on average, same total
+                             pixel values as the input image (True) or should the pixel values
+                             summed over all the images equal the sum of pixel values of the input
+                             image (False)? [default: False]
+    @param suppress_warnings Suppresses the warnings about the pixel scale of the output, if True.
+                             [default: False]
+
+    @returns a list of images and offsets to reconstruct the input image using 'interleaveImages'.
+    """
+
+    if isinstance(N,int):
+        n1,n2 = N,N
+    elif hasattr(N,'__iter__'):
+        if len(N)==2:
+            n1,n2 = N
+        else:
+            raise TypeError("'N' has to be a list or a tuple of two integers")
+        if not (isinstance(n1,int) and  isinstance(n2,int)):
+            raise TypeError("'N' has to be of type int or a list or a tuple of two integers")
+    else:
+        raise TypeError("'N' has to be of type int or a list or a tuple of two integers")
+
+    if not isinstance(image,galsim.Image):
+        raise TypeError("'image' has to be an instance of galsim.Image")
+
+    y_size,x_size = image.array.shape
+    if x_size%n1 or y_size%n2:
+        raise ValueError("The value of 'N' is incompatible with the dimensions of the image to "+
+                         +"be 'deinterleaved'")
+
+    im_list, offsets = [], []
+    for i in xrange(n1):
+        for j in xrange(n2):
+            # The tricky part - going from array indices to Image coordinates (x,y)
+            # DX[i'] = -(i+0.5)/n+0.5 = -i/n + 0.5*(n-1)/n
+            #    i  = -n DX[i'] + 0.5*(n-1)
```

Comment:
If I understand your concern correctly, the difference in the sign of the ```offset``` happens for the following reason. While calling ```drawImage```, when you provide an offset value of 0.25, it offsets the centroid of the profile by 0.25 pixels, which corresponds to moving the image boundary by -0.25, which is what you are seeing here. One can pass these offsets while calling ```drawImage``` and pass the resulting images to ```interleaveImage``` and get the original image back. I know this can get confusing, which is why I have some directional cases in my unit tests.